libyul/optimiser/UnusedStoreEliminator.cpp

```diff
@@ -0,0 +1,404 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Optimiser component that removes stores to memory and storage slots that are not used
+ * or overwritten later on.
+ */
+
+#include <libyul/optimiser/UnusedStoreEliminator.h>
+
+#include <libyul/optimiser/Semantics.h>
+#include <libyul/optimiser/OptimizerUtilities.h>
+#include <libyul/optimiser/Semantics.h>
+#include <libyul/optimiser/SSAValueTracker.h>
+#include <libyul/optimiser/DataFlowAnalyzer.h>
+#include <libyul/optimiser/KnowledgeBase.h>
+#include <libyul/ControlFlowSideEffectsCollector.h>
+#include <libyul/AST.h>
+
+#include <libsolutil/CommonData.h>
+
+#include <libevmasm/Instruction.h>
+#include <libevmasm/SemanticInformation.h>
+
+#include <range/v3/algorithm/all_of.hpp>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::yul;
+
+/// Variable names for special constants that can never appear in actual Yul code.
+static string const zero{"@ 0"};
```

Comment:
```suggestion
static string const zero{"0"};
```
should also be just fine... but doesn't matter... eventually it would probably be worth a thought to support implicit literal values in the knowledge base directly... but fine for now.